mitodl/ocw-studio
# Use webhooks and periodic task for updating publishing statuses

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #817 

#### What's this PR do?
- Adds webhooks back to the pipeline definitions.  The requests send back various build statuses: "started" at the beginning,  "succeeded" if it completes, "failed" if anything breaks.
- Adds a periodic task to query concourse for websites with pending builds whose statuses have not been updated in at least X (default 10) minutes.  If the status has not changed after a certain cutoff number of minutes (default 30) then the build is aborted.
- Tweaks the pipeline definition to skip requiring a git private key for public repos if `settings.CONCOURSE_IS_PRIVATE_REPO=True` (default is `False`).

#### How should this be manually tested?
- Use same AWS, CONCOURSE, GIT settings as RC
- Download ngrok and run `./ngrok http 8043`.  Copy the ngrok url to `OCW_STUDIO_BASE_URL=<ngrok_url>` in your .env file
- Restart docker instances
- Run `manage.py import_ocw_course_sites --filter res-7-001-pre-7-01-getting-up-to-speed-in-biology-summer-2019`
- Run `manage.py sync_backend_to_db --website res-7-001-pre-7-01-getting-up-to-speed-in-biology-summer-2019`
- Run `manage.py backpopulate_pipelines --filter  res-7-001-pre-7-01-getting-up-to-speed-in-biology-summer-2019`
- Publish a new draft version of the site from the front end.  Check that the status indicator updates as the build starts on concourse, and eventually changes to failed or completed.  You should also get an email at the end.
- When done testing, run this on heroku RC to get the pipeline back to what it should be: `manage.py backpopulate_pipelines --filter  res-7-001-pre-7-01-getting-up-to-speed-in-biology-summer-2019`
